### DISCUSSION TITLE
Doris可以同时使用动态分区和动态分桶吗？

### DISCUSSION THREAD

**Discussion by LCehoennardo** [apache/doris#29193] [A - General / Q&A]
如题，如果可以应该怎么使用呢，我是这么写的，但是报错了
```sql
......
PARTITION BY RANGE(statdate) ()
DISTRIBUTED BY HASH(statdate,app_id, order_id, order_goods_id) 
BUCKETS AUTO
PROPERTIES (
    "replication_allocation" = "tag.location.default: 3",
    "storage_format" = "V2",
    "light_schema_change" = 'true',
    "dynamic_partition.enable" = "true",
    "dynamic_partition.time_unit" = "DAY",
    "dynamic_partition.start" = "-7",
    "dynamic_partition.end" = "3",
    "dynamic_partition.prefix" = "p",
    "estimate_partition_size" = "170G"
);
```

报错信息很不明确，是：
```
Execution failed: Error Failed to execute sql: java.lang.Error: Error: could not match input
```
<img width="674" alt="image" src="https://github.com/apache/doris/assets/41560874/26818275-bfcc-4e9f-a3a1-db9a2b139637">

**Comment by LCehoennardo** [comment]
Doris version：1.2.5

**Reply by LemonLiTree** [reply]
1.2 not support BUCKETS AUTO

**Reply by LCehoennardo** [reply]
官网不是显示从1.2.2开始支持吗
<img width="654" alt="image" src="https://github.com/apache/doris/assets/41560874/24e4e4e0-d345-46d6-86f7-c1a1a3481f3f">

**Reply by LemonLiTree** [reply]
在mysql client试试呢

**Reply by LCehoennardo** [reply]
在命令行报的是这个错，依然看不出什么来：
<img width="841" alt="image" src="https://github.com/apache/doris/assets/41560874/001bc02f-29c7-4f51-8438-289e179a2f00">

如果不加`"estimate_partition_size" = "170G"`参数，就能正常创建，但无法起到动态分桶的效果，而是默认创建了10个分桶
<img width="895" alt="image" src="https://github.com/apache/doris/assets/41560874/b00168a6-8a47-4bd2-ad63-004d56599305">

是什么原因呢大佬

**Comment by LemonLiTree** [comment]
2.0 test is ok.

**Reply by LCehoennardo** [reply]
Does this mean this is a bug in version 1.2.5?

**Reply by LemonLiTree** [reply]
maybe...

**Reply by LCehoennardo** [reply]
Ok, thank you for your answer.
